pkgs/development/python-modules/grep-ast/default.nix

```diff
@@ -4,30 +4,33 @@
   lib,
 
   pathspec,
-  pytestCheckHook,
   setuptools,
-  tree-sitter-languages,
+  tree-sitter-language-pack,
 }:
 
-buildPythonPackage rec {
+let
+  version = "0.8.1";
```

Comment:
Why are you extracting this value to a let block?

Reply:
The let vs rec is a bit of a style preference to be honest. Personally i like the let better because you know exactly which variable is being ised, whereas with the rec, you have to check the code to understand what the rec is for exactly.

Reply:
I saw in the [best practices](https://nix.dev/guides/best-practices.html#recursive-attribute-set-rec) that using `rec` is considered an anti-pattern and should generally be avoided.

However, after revisiting issue #315337, I think it might be reasonable to keep using rec for now and wait for the treewide migration, to avoid unnecessary discussions around style preferences.